src/log_mgr.cc

```diff
@@ -1133,6 +1134,13 @@ Status LogMgr::doLogReclaim() {
     return Status();
 }
 
+Status LogMgr::doLogReclaimIfNecessary() {
+    if (numMemtables > getDbConfig()->maxKeepingMemtables) {
+        return doLogReclaim();
```

Comment:
Instead of this blocking call, we can do it in asynchronous way
```C++
DBMgr* db_mgr = DBMgr::getWithoutInit();
if (!db_mgr) return Status::NOT_INITIALIZED;
db_mgr->workerMgr()->invokeWorker("reclaimer");
```
Then we don't need to use `recursive_mutex`.

Reply:
Changed.  One cavaet on this, with background reclaim we will not evict memtables if there is no active reclaimer.  I think this behavior is ok.